### PR TITLE
DEP: Loosen dependency requirements

### PR DESCRIPTION
**Pull request by richford**
Inspired by [this pyAFQ PR](https://github.com/yeatmanlab/pyAFQ/pull/636), this PR loosens groupyr's dependency requirements.

### CODE REVIEW

**Comment by richford** [comment]
Okay, this now relaxes a lot of other dependencies and uses `tox` to test two different sklearn versions. It turns out our sklearn version requirements are narrow (just 0.23.1 and 0.23.2). We're limited on the lower end by using the private `_validate_data` method, which appeared in 0.23.1, and on the upper end by the skopt bug referenced in #47.

Other than that, I think we're good to go here.

File: groupyr/logistic.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def predict(self, X):
         """
         scores = self.decision_function(X)
         if len(scores.shape) == 1:
-            indices = (scores > 0).astype(np.int)
+            indices = (scores > 0).astype(np.int32)
```
Comment (arokem) [inline groupyr/logistic.py:176]: Is this a Python 3.9 thing?

**Review by arokem — COMMENTED** [review]
Yep. Looks good!

File: groupyr/logistic.py (hunk 1 above)
Comment (richford) [reply groupyr/logistic.py:176]: It's a numpy thing. I was getting warnings about `np.int` and `np.float` being deprecated in favor of either the build in Python `int` and `float` or the specific `np.int32` and `np.float64`.

**Pull request merged**